framework/include/media/MediaRecorderObserverInterface.h

```diff
@@ -53,25 +59,37 @@ class MediaRecorderObserverInterface
 	 * @details @b #include <media/MediaRecorderObserverInterface.h>
 	 * @since TizenRT v2.0 PRE
 	 */
-	virtual void onRecordStarted(Id id) = 0;
+	virtual void onRecordStarted(MediaRecorder& mr) = 0;
 	/**
 	 * @brief informs the user of the recording has paused.
 	 * @details @b #include <media/MediaRecorderObserverInterface.h>
 	 * @since TizenRT v2.0 PRE
 	 */
-	virtual void onRecordPaused(Id id) = 0;
+	virtual void onRecordPaused(MediaRecorder& mr) = 0;
 	/**
 	 * @brief informs the user of the recording has finished.
 	 * @details @b #include <media/MediaRecorderObserverInterface.h>
 	 * @since TizenRT v2.0 PRE
 	 */
-	virtual void onRecordFinished(Id id) = 0;
+	virtual void onRecordFinished(MediaRecorder& mr) = 0;
+	/**
+	 * @brief informs the user of the error state of recorder start operation
+	 * @details @b #include <media/MediaRecorderObserverInterface.h>
+	 * @since TizenRT v2.0 PRE
+	 */
+	virtual void onRecordStartError(MediaRecorder& mr, int errCode) = 0;
```

Comment:
Actually recorder_result_e is correct, instead of int.
But 'result' is not good option for error code.
Below are suggestion.
1. Rename recorder_result_e to recorder_error_e
2. Rename RECORDER_OK to RECORDER_ERROR_NONE